### PR TITLE
Show Wikipedia Previews on Wikipedia hyperlinks

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link : https://phabricator.wikimedia.org/T250850

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
...
+          The <span class="wmf-wp-with-preview" data-wikipedia-preview>cat</span> (Felis catus) is a small <span class="wmf-wp-with-preview" data-wikipedia-preview>carnivorous</span> mammal. It is the only domesticated species in the family <a href="https://en.wikipedia.org/wiki/Felidae">Felidae</a> and often referred to as the domestic cat to distinguish it from wild members of the <span class="wmf-wp-with-preview" data-wikipedia-preview>family</span>. The cat is either a house cat or a farm cat, which are pets, or a feral cat, which ranges freely and avoids human contact. A house cat is valued by <a href="https://en.wikipedia.org/wiki/humans">humans</a> for companionship and <span class="wmf-wp-with-preview" data-wikipedia-preview>for (not an article)</span> its ability to hunt rodents. About 60 cat breeds are recognized by various cat registries.
```
Comment (hueitan) [inline index.html:23]: this is the temporarily a tag solution for this demo purpose, ultimately this demo page will be replaced with another demo page.

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const getWikipediaAttrFromUrl = url => {
+	const matches = /^https?:\/\/(\w+)(\.m)?\.wikipedia\.org\/wiki\/(\w+)/.exec( url )
```
Comment (stephanebisson) [inline src/utils.js:2]: If an editor goes to the site and copies the URL from the browser, they may end up with something like `https://en.wikipedia.org/w/index.php?title=Cat`, that should also be supported.

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export const getWikipediaAttrFromUrl = url => {
```
Comment (stephanebisson) [inline src/utils.js:1]: Please add unit tests for this util

**Comment by medied** [comment]
Looking pretty good, we should probably update the README to reflect this new functionality

**Comment by stephanebisson** [comment]
Looks good. Should be good to go after updating the README

**Comment by hueitan** [comment]
TODO : This needs to be tested with Touch event, touch the hyperlink opens the popup instead of the a link.

File: README.md [hunk 4]
```diff
@@ -73,6 +74,19 @@ wikipediaPreview.init({
 });
 ```
 
+Example (detect wikipedia link)
```
Comment (stephanebisson) [inline README.md:77]: **W**ikipedia link**s**

File: README.md [hunk 5]
```diff
@@ -55,8 +55,9 @@ root | DOM Element | `document` | Where to look for elements that should have th
 selector | string | `'[data-wikipedia-preview]'` | How nodes that should have the popup are identified
 lang | string | `'en'` | Default Wikipedia language
 popupContainer | DOM Element | `document.body` | Where to put the popup in the DOM
+detectLinks | Boolean | `false` | Allow Wikipedia hyperlink to have the popup
```
Comment (stephanebisson) [inline README.md:58]: hyperlink**s**

**Comment by stephanebisson** [comment]
> TODO : This needs to be tested with Touch event, touch the hyperlink opens the popup instead of the a link.

Touch interaction is not yet on master so you can safely ignore it here. If it gets merged first, you'll have to make sure this PR works well with it.

**Comment by hueitan** [comment]
> > TODO : This needs to be tested with Touch event, touch the hyperlink opens the popup instead of the a link.
> 
> Touch interaction is not yet on master so you can safely ignore it here. If it gets merged first, you'll have to make sure this PR works well with it.

So after the discussion, mobile support is our priority, we will wait until the touch interaction is done https://github.com/wikimedia/wikipedia-preview/pull/18, then make sure the Wikipedia Hyperlinks works as well in the mobile browser within this PR.

**Comment by hueitan** [comment]
This branch is updated with the latest mobile interaction, therefore it's ready for review again

**Comment by jpita** [comment]
I need help figuring out how this works.
I tried to put this code on `index.html` 


```<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
    />
    <link href="dist/wikipedia-preview.css" rel="stylesheet">
    <style>
      .content {
        margin: 20px 10%;
      }
    </style>
    <title>Wikipedia Preview demo</title>
  </head>

  <body>
  	<div >
    	<p class="content">
        You can learn about <a href="https://en.wikipedia.org/wiki/Chat">Chat</a> and <a href="https://en.wikipedia.org/wiki/Chien">Chien</a> from Wikipedia.
      </p>
		<div class="popup-container"></div>
    </div>
    <script src="dist/wikipedia-preview.development.js"></script>
    <script type="text/javascript">
	    wikipediaPreview.init({
				detectLinks: true
		});
    </script>
  </body>
</html>
``` 

and this is the result I get


image

**Comment by hueitan** [comment]
Yeah u got that

**Pull request merged**